src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -292,8 +336,11 @@ private CommandResult executeCommand(String commandText) throws CommandException
 
             if (commandResult.getMode() != null) {
                 switchMode(commandResult.getMode());
+            } else {
+                updateDetailsPanel();
```

Comment:
might be good to have some documentation to make it easier to understand 
e.g. explaining why updateDetailsPanel() is only invoked when a non SwitchCommand is invoked and how the UI is updated when